lib/message_handler.rb

```diff
@@ -11,43 +11,62 @@ class MessageHandler < FeatureFlag
   release_for 'Fabien Townsend'
 
   def initialize(args = {})
-    @mark_all_out = args[:mark_all_out]
+    @mark_all_out = args[:mark_all_out] # TODO - this doesnt seem to be used here.
     @request_parser = RequestParser.new
     @bot = args[:bot]
     @user_info = args[:user_info_provider]
   end
 
   def handle(event_data)
-    recipient = event_data['channel'] || event_data['user']
-    data = format_data(event_data)
-    return data[:user_message] if data[:user_message].nil?
-    returned_command = @request_parser.parse(data)
+    data = format(event_data)
 
-    unless User.profile(data[:user_id])
+    if has_no_message?(data)
+      return data[:user_message]
+    end
+
+    if has_no_user?(data)
```

Comment:
I would rather stick with positive assertion through the application and whenever need the `no` use either `!` or `unless`